{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bb27a273-a5c9-41c0-b568-ebb346c15afa",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext sql"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f9992408-376a-43d1-9101-2654b8b41a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv, sqlite3\n",
    "\n",
    "conn = sqlite3.connect('SQLMagic.db')\n",
    "cur = conn.cursor()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f57a7d61-30b1-4039-a676-6eb0225c5ca0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sql\\connection.py\", line 45, in __init__\n",
      "    engine = sqlalchemy.create_engine(\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\engine\\__init__.py\", line 525, in create_engine\n",
      "    return strategy.create(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\engine\\strategies.py\", line 61, in create\n",
      "    entrypoint = u._get_entrypoint()\n",
      "                 ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 172, in _get_entrypoint\n",
      "    cls = registry.load(name)\n",
      "          ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\util\\langhelpers.py\", line 277, in load\n",
      "    raise exc.NoSuchModuleError(\n",
      "sqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:sqlite3\n",
      "\n",
      "Connection info needed in SQLAlchemy format, example:\n",
      "               postgresql://username:password@hostname/dbname\n",
      "               or an existing connection: dict_keys([])\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sql\\magic.py\", line 196, in execute\n",
      "    conn = sql.connection.Connection.set(\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sql\\connection.py\", line 70, in set\n",
      "    cls.current = existing or Connection(descriptor, connect_args, creator)\n",
      "                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sql\\connection.py\", line 45, in __init__\n",
      "    engine = sqlalchemy.create_engine(\n",
      "             ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\engine\\__init__.py\", line 525, in create_engine\n",
      "    return strategy.create(*args, **kwargs)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\engine\\strategies.py\", line 61, in create\n",
      "    entrypoint = u._get_entrypoint()\n",
      "                 ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\engine\\url.py\", line 172, in _get_entrypoint\n",
      "    cls = registry.load(name)\n",
      "          ^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\shvmpz\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\sqlalchemy\\util\\langhelpers.py\", line 277, in load\n",
      "    raise exc.NoSuchModuleError(\n",
      "sqlalchemy.exc.NoSuchModuleError: Can't load plugin: sqlalchemy.dialects:sqlite3\n",
      "\n",
      "Connection info needed in SQLAlchemy format, example:\n",
      "               postgresql://username:password@hostname/dbname\n",
      "               or an existing connection: dict_keys([])\n"
     ]
    }
   ],
   "source": [
    "%sql sqlite3:///SQLMagic.db"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e4318730-babb-4326-a414-00ca27b6aee4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
